fusionauth/resource_fusionauth_generic_connector.go

```diff
@@ -0,0 +1,250 @@
+		return diag.FromErr(err)
+	}
+	data.SetId(resp.Connector.Id)
+	return nil
+}
+
+func readGenericConnector(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	id := data.Id()
+
+	resp, faErrs, err := retrieveConnector(*client.FAClient, id, GenericConnectorRequest{})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		data.SetId("")
+		return nil
+	}
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+
+	connector := resp.Connector
+	if err := data.Set("authentication_url", connector.AuthenticationURL); err != nil {
+		return diag.Errorf("connector.authentication_url: %s", err.Error())
+	}
+	if err := data.Set("connect_timeout", connector.ConnectTimeout); err != nil {
+		return diag.Errorf("connector.connect_timeout: %s", err.Error())
+	}
+	if err := data.Set("data", connector.Data); err != nil {
+		return diag.Errorf("connector.data: %s", err.Error())
+	}
+	if err := data.Set("debug", connector.Debug); err != nil {
+		return diag.Errorf("connector.debug: %s", err.Error())
+	}
+	if err := data.Set("headers", connector.Headers); err != nil {
+		return diag.Errorf("connector.headers: %s", err.Error())
+	}
+	if err := data.Set("http_authentication_password", connector.HttpAuthenticationPassword); err != nil {
+		return diag.Errorf("connector.http_authentication_password: %s", err.Error())
+	}
+	if err := data.Set("http_authentication_username", connector.HttpAuthenticationUsername); err != nil {
+		return diag.Errorf("connector.http_authentication_username: %s", err.Error())
+	}
+	if err := data.Set("name", connector.Name); err != nil {
+		return diag.Errorf("connector.name: %s", err.Error())
+	}
+	if err := data.Set("read_timeout", connector.ReadTimeout); err != nil {
+		return diag.Errorf("connector.read_timeout: %s", err.Error())
+	}
+	if err := data.Set("ssl_certificate_key_id", connector.SslCertificateKeyId); err != nil {
+		return diag.Errorf("connector.ssl_certificate_key_id: %s", err.Error())
+	}
+
+	return nil
+}
+
+func updateGenericConnector(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	connector := buildGenericConnector(data)
+
+	resp, faErrs, err := updateConnector(*client.FAClient, data.Id(), GenericConnectorRequest{Connector: connector})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func deleteGenericConnector(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	id := data.Id()
+
+	resp, faErrs, err := client.FAClient.DeleteConnector(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+type GenericConnectorRequest struct {
```

Comment:
The fusion auth go client can't create, read or update generic connectors as all of the connector methods use the `BaseConnectorConfiguration` type. I've looked at the go client code and copied across the logic for managing connectors and adapted it to work for the `GenericConnectorConfiguration` type.